tfx/orchestration/beam/beam_dag_runner.py

```diff
@@ -93,6 +106,22 @@ def run(self, tfx_pipeline: pipeline.Pipeline) -> None:
     """
     # For CLI, while creating or updating pipeline, pipeline_args are extracted
     # and hence we avoid deploying the pipeline.
+
+    # Google Dataflow restricts naming to only alphanumeric and dashes
+    job_name = re.sub(
+      r'[^0-9a-zA-Z-]+',
+      '-',
+      '{pipeline_name}-{ts}'.format(
+        pipeline_name=tfx_pipeline.pipeline_info.pipeline_name,
+        ts=int(datetime.datetime.timestamp(datetime.datetime.now()))
+      )
+    )
+    # Exploit the last-in priority of --job_name for beam
+    if self._beam_orchestrator_args:
+      self._beam_orchestrator_args.append('--job_name={}'.format(job_name))
+    else:
+      self._beam_orchestrator_args = ['--job_name={}'.format(job_name)]
```

Comment:
the beam orchestrator is for entire pipeline instead of a single component

Reply:
Hi @1025KB , maybe I have a misunderstanding then and you can help me by clarifying. I was under the assumption from the comments in the code that `self._beam_orchestrator_args` are responsible for the orchestrator pipeline args and that `component_launcher.ComponentLauncher()` takes `additional_pipeline_args`, which in turn contain a list of `beam_pipeline_args`. This drove me to these design decisions.

Please also note that the reviewed code contained a bug under which the actual pipeline names for executors could be wrong due to a race-condition-like flaw in my own naming logic. According to my tests its fixed now.